### ISSUE TITLE
Labels on edge connections

### ISSUE THREAD

**Issue by jabcross** [mermaid-js/mermaid#1652]
I'd like a feature to allow something like this.

![2020-08-27-042759_300x103_scrot](https://user-images.githubusercontent.com/6402237/91410954-0fcd5800-e81e-11ea-922f-b0b2e53bbf58.png)

(ignore the (a), it's a LaTeX label)

Right now, edges only allow a single label, in the center. I'd like to be able to label the connection between edges and nodes, as well.

**Comment by Opiboble** [comment]
Yeah this would be amazing to have. I am currently using Mermaid to help with network documentation and this would make labeling connecting ports cleaner.

Something like:

`A |port1|---|port2|B`

**Comment by AkechiShiro** [comment]
What is blocking this issue, no one is working on it ?

**Comment by sidharthv96** [comment]
We'd have to figure out a syntax that doesn't break the existing diagrams. 

```
// Works when both sides have labels, but if what should we do when we need a label on both ends, and the center?
A|port1|-->|port2|B

// Where should the label be placed now? To the right, or the center (current behaviour).
A-->|port2|B
```

**Comment by DanEble** [comment]
B=Beginning, M=Middle, E=End, A=Automatic
```
|B|---|M|-->|E|
|B|---|M|-->||
|B|---||-->|E|     |B|-->|E|
|B|---||-->||      |B|-->||
 ||---|M|-->|E|
 ||---|M|-->||                  ---|M|-->    -->|A|
 ||---||-->|E|      ||-->|E|
 ||---||-->||       ||-->||                  -->
```

**Comment by antonsizikov** [comment]
I lack this feature too. I think that syntax @DanEble wrote doesn't break the existing diagrams and is really obvious for understanding. Will wait for someone to contribute on it.

**Comment by jabcross** [comment]
Just a comment: [Elk supports ports.](https://rtsys.informatik.uni-kiel.de/elklive/examples.html?e=general%2Fspacing%2Fports). Now that Mermaid supports Elk as an alternate renderer, I think this should be straightforward to implement.
